pkg/cloud/services/network/subnets.go

```diff
@@ -262,6 +267,73 @@ func (s *Service) deleteSubnets() error {
 	return nil
 }
 
+func (s *Service) deleteAWSCloudProviderTags() error {
+	out, err := s.describeSubnetsByAWSCloudProviderTag()
+	if err != nil {
+		return err
+	}
+	if len(out.Subnets) == 0 {
+		return nil
+	}
+
+	s.scope.V(4).Info("Deleting AWSCloudProvider tags from unmanaged subnets")
+
+	var publicSns []string
+	var privateSns []string
+	for _, sn := range out.Subnets {
+		id := aws.StringValue(sn.SubnetId)
+		// Check if the subnet is in AWSCluster spec
+		if s.scope.Subnets().FindByID(id) == nil {
+			continue
+		}
+
+		if *sn.MapPublicIpOnLaunch {
+			publicSns = append(publicSns, id)
+		} else {
+			privateSns = append(privateSns, id)
+		}
+	}
+
+	if err := s.deleteTags(publicSns, s.getAWSProviderTags(true)); err != nil {
+		return err
+	}
+	if err := s.deleteTags(privateSns, s.getAWSProviderTags(false)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *Service) deleteTags(resourceIDs []string, tags infrav1.Tags) error {
+	if len(resourceIDs) == 0 {
+		return nil
+	}
+
+	// Create the DeleteTags input
+	input := &ec2.DeleteTagsInput{
+		Resources: aws.StringSlice(resourceIDs),
+		Tags:      converters.MapToTags(tags),
+	}
+
+	if _, err := s.EC2Client.DeleteTags(input); err != nil {
+		return errors.Wrapf(err, "failed to delete AWSCloudProvider tags from unmanaged subnets %q", resourceIDs)
+	}
+
+	return nil
+}
+
+func (s *Service) getAWSProviderTags(public bool) infrav1.Tags {
+	t := infrav1.Tags{}
+
+	t[infrav1.ClusterAWSCloudProviderTagKey(s.scope.KubernetesClusterName())] = string(infrav1.ResourceLifecycleShared)
+	if public {
+		t[externalLoadBalancerTag] = "1"
```

Comment:
What if the unmanaged VPC is used to create multiple clusters? 
With deleting internal/external LB tags, we will disrupt those clusters' CCM operations until the next reconcile where the tags are readded.

Leaving them is also bad, how about deleting the LB tag only if there is no other CCM tag that belongs to any other cluster?